apollo-router/tests/lifecycle_tests.rs

```diff
@@ -79,3 +85,61 @@ async fn create_router(config: &str) -> Result<IntegrationTest, BoxError> {
 
     Ok(IntegrationTest::new(tracer, opentelemetry_jaeger::Propagator::new(), config).await)
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_graceful_shutdown() -> Result<(), BoxError> {
+    let tracer = opentelemetry_jaeger::new_agent_pipeline()
+        .with_service_name("my_app")
+        .install_simple()?;
+
+    let mut router = IntegrationTest::with_mock_responder(
+        tracer,
+        opentelemetry_jaeger::Propagator::new(),
+        "telemetry:
+  tracing:
+    propagation:
+      jaeger: true
+    trace_config:
+      service_name: router
+    jaeger:
+      batch_processor:
+        scheduled_delay: 100ms
+      agent:
+        endpoint: default
+override_subgraph_url:
+  products: http://localhost:4005
+include_subgraph_errors:
+  all: true
+",
+       TracedResponder(ResponseTemplate::new(200).set_body_json(
+            json!({"data":{"topProducts":[{"name":"Table"},{"name":"Couch"},{"name":"Chair"}]}}),
+        ).set_delay(Duration::from_secs(5))),
+    )
+    .await;
+
+    router.start().await;
+    router.assert_started().await;
+    let pid = router.pid();
+
+    let client_handle = tokio::task::spawn(async move {
+        let (_, res) = router.run_query().await;
+        (
+            router,
+            serde_json::from_slice::<graphql::Response>(&res.bytes().await.unwrap()).unwrap(),
+        )
+    });
+
+    tokio::time::sleep(Duration::from_millis(1000)).await;
+    unsafe {
+        libc::kill(pid, libc::SIGINT);
```

Comment:
Just making sure -- it will also work with SIGTERM, right?

Reply:
the router considers both SIGINT and SIGTERM: https://github.com/apollographql/router/blob/dev/apollo-router/src/router.rs#L614-L623